spec/integration/data_stream/manifest.spec.yml

```diff
@@ -75,6 +76,27 @@ spec:
             default: false
             examples:
               - true
+          options:
+            description: List of options for select type
+            type: array
+            items:
+              type: object
+              additionalProperties: false
+              properties:
+                value:
+                  type: string
+                  examples:
+                  - node
+                  - cluster
+                text:
+                  type: string
+                  examples:
+                  - node
+                  - cluster
+              required:
+              - value
+              - text
+            default: []
```

Comment:
As this options property should just be defined if the `select` type is chosen, it could be tried to use the if-then-else of json-schema to force that:
https://json-schema.org/understanding-json-schema/reference/conditionals.html#if-then-else

If that does not work, we would need to add a new validate rule for that. There are some similar validator rules in place now , for instance:
https://github.com/elastic/package-spec/blob/main/code/go/internal/validator/semantic/validate_field_groups.go
If a new rule is added, it should be added here too: https://github.com/elastic/package-spec/blob/main/code/go/internal/validator/spec.go#L86